ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/views/imagebrowser/repositories/AbstractRepository.java

```diff
@@ -45,7 +45,7 @@
 
 public abstract class AbstractRepository extends Job {
 
-	protected List<ImageElement> mElementsCache = new LinkedList<>();
+	private Map<String, ImageElement> mElementsCache = new LinkedHashMap<>();
```

Comment:
```suggestion
	private SequencedMap<String, ImageElement> mElementsCache = new LinkedHashMap<>();
```
Then casting the field to `LinkedHashMap` below should also not be necessary.